### DISCUSSION TITLE
Is there a way to mock and test the AutomationCompatible Upkeep?

### DISCUSSION THREAD

**Discussion by Vincent71399** [Cyfrin/foundry-full-course-cu#2197] [Q&A]
I think in the video it only mocked the vrf, but I found mock dir under automation also, is there a way to mock that in the test?

**Comment by EngrPips** [comment]
Hello @Vincent71399, I think we directly trigger the `checkUpKeep` and `performUpKeep` functions in our test, which in itself is a way of mocking the automation system. But mocking them in a way that we set up a node that automatically calls our contract can be too much of a way I think.
